cli/flox/src/commands/environment.rs

```diff
@@ -307,21 +314,12 @@ impl Activate {
 
         let mut environment = concrete_environment.into_dyn_environment();
 
-        let mut stderr = std::io::stdout();
-        stderr
-            .queue(cursor::SavePosition)
-            .context("couldn't set cursor positon")?;
-        stderr
-            .write_all("Building environment...\n".as_bytes())
-            .context("could't write progress message")?;
-        stderr.flush().context("could't flush stderr")?;
-
-        let activation_path = environment.activation_path(&flox)?;
-
-        stderr
-            .queue(cursor::RestorePosition)
-            .context("couldn't restore cursor position")?;
-        stderr.flush().context("could't flush stderr")?;
+        let activation_path = Dialog {
+            message: &format!("Building environment '{prompt_name}'..."),
```

Comment:
note: we should probably use `environment_description()` instead of `prompt_name`